src/livekit/useLiveKit.ts

```diff
@@ -19,6 +25,10 @@ type LiveKitState = {
   room: Room;
 };
 
+function emptyToUndef(str) {
+  return str === "" ? undefined : str;
+}
+
```

Comment:
Oh, when I meant moving things to a function, I meant that thing that selects an active device based on the active device setting and the default device setting, not the conversion to undefined 🙂

Reply:
I was considering this. But since we have `activeAudioDevice`, `settingsDefaultDevices.audioinput` `audioDevices[0]?.deviceId` there are 3 parameters that need to be different for each function call. So it is not really possible to make it very generic. I would end up to be a function that looks like so:
```
function (a,b,c){
a === "" ? (b === "" ? c : b) : a
)
```
which i found less readable than `??` chaingin. WIth `??` chaing it is at least obvious what priority the different variables have.
wdyt?